### PR TITLE
Patch article

### PR DESCRIPTION
**Pull request by Kamz1991**

### CODE REVIEW

File: __tests__/project.test.js [hunk 1]
```diff
@@ -57,3 +57,50 @@ describe("GET /api/articles/article:id", () => {
...
+    };
+    return request(app)
+      .patch("/api/articles/1")
+      .send(articleUpdates)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("bad request");
+      });
+  });
+});
```
Comment (xmenbteam) [inline __tests__/project.test.js:118]: What about a 404 article not found?

File: controllers/controller.js [hunk 2]
```diff
@@ -1,4 +1,8 @@
-const { selectTopics, selectAuthorById } = require("../models/model");
+const {
+  selectTopics,
+  selectAuthorById,
+  updateArticlesById,
+} = require("../models/model");
```
Comment (xmenbteam) [inline controllers/controller.js]: You should have SEPARATE files for topics and authors models. Have a look at my 'naming conventions' file that I posted in the cohort channel yesterday.

File: models/model.js [hunk 3]
```diff
@@ -17,3 +17,18 @@ exports.selectAuthorById = async (article_id) => {
...
+exports.updateArticlesById = async (article_id, newVotes) => {
+  const result = await db.query(
+    `UPDATE articles 
+   SET votes = votes + $1 
+    WHERE article_id = $2 
+    RETURNING *;`,
+    [newVotes, article_id]
+  );
+  if (!result) {
+    return Promise.reject({ status: 400, msg: "bad request" });
```
Comment (xmenbteam) [inline models/model.js]: This isn't where you handle a 400, this is a psql error. Your 404 should be handled here, and it should be if(!result.rows[0]) that you check for.

**Review by xmenbteam — CHANGES_REQUESTED** [review]
Great work, but PLEASE refactor according to the guidelines I sent yesterday. Half an hour today will save a whole bucket of feelings later on in the week!

File: __tests__/project.test.js [hunk 4]
```diff
@@ -57,3 +57,62 @@ describe("GET /api/articles/article:id", () => {
...
+      .patch("/api/articles/1")
+      .send(articleUpdates)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("bad request");
+      });
+  });
+  test("invalid inc votes like a string instead of a number", () => {
+    const articleUpdates = {
+      inc_votes: "dog",
```
Comment (dentednerd) [inline __tests__/project.test.js:96]: I like that you've tested for both the absence of `inc_votes` and an invalid value of `inc_votes`. Nice work!

**Review by dentednerd — APPROVED** [review]
Good job addressing Sam's feedback, this is looking good now!

File: __tests__/project.test.js (hunk 4 above)
Comment (Kamz1991) [reply __tests__/project.test.js:96]: thanks

**Pull request merged**